docs/fsharp/tour.md

```diff
@@ -1,4 +1,158 @@
+
+[!code-fsharp[Records](../../samples/snippets/fsharp/tour.fs#L357-L373)]
+
+[Discriminated Unions](language-reference/discriminated-unions.md) are values which could be a number of named forms or cases.  Data stored in the type can be one of several distinct values.
+
+[[!code-fsharp[Unions](../../samples/snippets/fsharp/tour.fs#L381-L432)]
+
+You can also use Unions as *single-case unions*, to help with domain modeling over primitive types.  Often times, strings and other primitive types are used to represent types of information that shouldn't be interchangeable.  However, using only the string representation can lead to that mistake quite easily!  Representing each type of information as a distinct single-case union can enforce correctness in this scenario.
+
+[[!code-fsharp[Unions](../../samples/snippets/fsharp/tour.fs#L434-L437)]
+
+Additionally, Discriminated Unions also support recursive definitions, allowing you to easily represent trees and inherently recursive data.  For example, here's how you can represent a Binary Search Tree with `exists` and `insert` functions.
+
+[[!code-fsharp[Unions](../../samples/snippets/fsharp/tour.fs#L439-L466)]
+
+Because Discriminated Unions allow you to represent the recursive structure of the tree in the data type, operating on it is very straightforward and nets you correctness guarantees with pattern matching.  See below for more.
+
+## Pattern Matching
+
+[Pattern Matching](language-reference/pattern-matching.md) is the F# languag feature which enables correctness for operating on F# types.  In the above samples, you probably noticed quite a bit of `match x with ...` syntax.  This construct allows the compiler, which can understand the "shape" of data types, to enforce you to account for all possible cases when using a data type through what is known as Exhausting Pattern Matching.  This is incredibly powerful for correctness, and can be cleverly used to "lift" what would normally be a runtime concern into compile-time.
+
+[[!code-fsharp[PatternMatching](../../samples/snippets/fsharp/tour.fs#L496-L528)]
+
+You can also use the shorthand `function` construct for pattern matching, which is useful when you're writing functions which make use of [Partial Application](language-reference/functions/index.md#partial-application-of-arguments):
+
+[[!code-fsharp[PatternMatching](../../samples/snippets/fsharp/tour.fs#L530-L547)]
+
+Something you may have noticed is the use of the `_` pattern.  This is known as the "wildcard" pattern, which is basically a way of saying "I don't care what it is, since it doesn't matter to me".  Although convenient, you can accidentaly bypass Exhaustive Pattern Matching and no longer benefit from compile-time enforcements if you aren't careful.
```

Comment:
Best bet would be to link to the specific section of the Pattern Matching topic.  I'll do that.